radicle-surf/src/source/object/tree.rs

```diff
@@ -126,16 +146,23 @@ impl Serialize for TreeEntry {
     {
         const FIELDS: usize = 4;
         let mut state = serializer.serialize_struct("TreeEntry", FIELDS)?;
-        state.serialize_field("path", &self.entry.location())?;
-        state.serialize_field("name", &self.entry.name())?;
-        state.serialize_field("lastCommit", &None::<commit::Header>)?;
+        state.serialize_field("name", &self.name)?;
         state.serialize_field(
             "kind",
             match self.entry {
-                directory::Entry::File(_) => "blob",
-                directory::Entry::Directory(_) => "directory",
+                Entry::Blob(_) => "blob",
+                Entry::Tree(_) => "tree",
             },
         )?;
+        state.serialize_field(
+            "oid",
+            match self.entry {
+                Entry::Blob(id) => id.to_string(),
+                Entry::Tree(id) => id.to_string(),
+            }
+            .as_str(),
```

Comment:
I'd suggest adding `fn object_id` to `TreeEntry`, and `ext::Oid` already implements `Serialize` 
```suggestion
            self.object_id()
```